client/cypress/end-to-end/batch-comparison-failure-cases.spec.js

```diff
@@ -0,0 +1,153 @@
+        mimeType: 'csv'
+      })
+    })
+    cy.findByText('Upload File').click()
+    cy.contains("Upload successfully completed")  
+
+    cy.get('button[type="submit"]').should('not.have.class', 'bp3-disabled').click()
+    cy.findAllByText('Target Contests').should('have.length', 2)
+    cy.get('input[name="contests[0].name"]').type('Contest')
+    cy.findByLabelText('Name of Candidate/Choice 1').type('Vader')
+    cy.findByLabelText('Votes for Candidate/Choice 1').type('2700')
+    cy.findByLabelText('Name of Candidate/Choice 2').type('Palpatine')
+    cy.findByLabelText('Votes for Candidate/Choice 2').type('2620')
+    cy.findByLabelText('Total Ballots for Contest').type('5320')
+    cy.findByText('Select Jurisdictions').click()
+    cy.findByLabelText('Death Star').check({ force: true })
+    cy.findByText('Save & Next').click()
+    cy.findAllByText('Audit Settings').should('have.length', 2)
+    cy.get('#state').select('AL')
+    cy.get('input[name=electionName]').type(`Test Election`)
+    cy.get('#risk-limit').select('10')
+    cy.get('input[name=randomSeed]').type("54321")
+    cy.findByText('Save & Next').click()
+    cy.findAllByText('Review & Launch').should('have.length', 2)
+    cy.logout(auditAdmin)
+    cy.loginJurisdictionAdmin(jurisdictionAdmin)
+    cy.findByText(`Jurisdictions - TestAudit${id}`).siblings('button').click()
+
+    // upload invalid manifest
+    cy.fixture('CSVs/manifest/batch_comparison_manifest_col_error.csv').then(fileContent => {
+      cy.get('input[type="file"]').first().attachFile({
+        fileContent: fileContent.toString(),
+        fileName: 'batch_comparison_manifest_col_error.csv',
+        mimeType: 'csv'
+      })
+    })
+    cy.findAllByText('Upload File').spread((firstButton, secondButton) => {
+      firstButton.click()
+    })
+    cy.contains("Missing required column: Number of Ballots.")
+
+    // upload valid manifest
+    cy.findByText('Replace File').click()
+    cy.findAllByText('Upload File').should('have.length',2)
+    cy.fixture('CSVs/manifest/batch_comparison_manifest.csv').then(fileContent => {
+      cy.get('input[type="file"]').first().attachFile({
+        fileContent: fileContent.toString(),
+        fileName: 'batch_comparison_manifest.csv',
+        mimeType: 'csv'
+      })
+    })
+    cy.findAllByText('Upload File').spread((firstButton, secondButton) => {
+      firstButton.click()
+    })
+    cy.contains("Upload successfully completed")
+
+
+
+    // upload invalid batch tallies
+    cy.fixture('CSVs/candidate-total-batch/sample_candidate_totals_by_batch_col_error.csv').then(fileContent => {
+      cy.get('input[type="file"]').last().attachFile({
+        fileContent: fileContent.toString(),
+        fileName: 'sample_candidate_totals_by_batch_col_error.csv',
+        mimeType: 'csv'
+      })
+    })
+    cy.findAllByText('Upload File').last().click()
+    cy.contains(/Missing required column: Palpatine/)
+    
```

Comment:
I think the test could probably end here, since there are no further error cases to check

Reply:
Makes sense. I've updated it, but decided to cut it off after the replace file process since that's technically part of the process of dealing with bad files, and the integration there isn't able to be effectively covered with the unit tests.